plugin-modernizer-core/src/test/java/io/jenkins/tools/pluginmodernizer/core/recipes/MergeGitIgnoreRecipeTest.java

```diff
@@ -0,0 +1,68 @@
+package io.jenkins.tools.pluginmodernizer.core.recipes;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.test.RewriteTest;
+import static org.openrewrite.test.SourceSpecs.text;
+import java.nio.file.Path;
+
+public class MergeGitIgnoreRecipeTest implements RewriteTest {
+    @Test
+    void shouldMergeGitIgnoreEntries() {
+        rewriteRun(
+            spec -> spec.recipe(new MergeGitIgnoreRecipe(Path.of("archetypes/common-files/.gitignore"))),
+            text(
+                """
+                # Existing user-defined entries
+                *.log
+                build/
+                .idea/
+                
+                # Custom section
+                custom/*.tmp
+                """,
+                sourceSpecs -> sourceSpecs.path(".gitignore")
+            ),
+            text(
+                """
+                target
+                work
+                
+                # mvn hpi:run
+                # IntelliJ IDEA project files
+                *.iml
+                *.iws
+                *.ipr
+                .idea
+                
+                # Eclipse project files
+                .settings
+                .classpath
+                .project
+                """,
+                sourceSpecs -> sourceSpecs.path("archetypes/common-files/.gitignore")
```

Comment:
This should not be part of the source files. I think the text file can be directly added as a constant on the recipe